### ISSUE TITLE
java.lang.ClassCastException :io.jenkins.docker.DockerComputer cannot be cast to hudson.slaves.AbstractCloudComputer

### ISSUE THREAD

**Issue by alexeygrigorovaurea** [jenkinsci/docker-plugin#569]
HI team

after we upgraded from 1.0.4 to 1.1 now our server keep  unsuccessfully provision nodes in infinite loop 
tried on jenkins ver 2.73.2 and 2.89.1 lts both same issue.
Exception appearing in log:
```
Timer task hudson.slaves.ComputerRetentionWork@528d4c34 failed
java.lang.ClassCastException: io.jenkins.docker.DockerComputer cannot be cast to hudson.slaves.AbstractCloudComputer
	at hudson.slaves.CloudRetentionStrategy.check(CloudRetentionStrategy.java:43)
	at hudson.slaves.ComputerRetentionWork$1.run(ComputerRetentionWork.java:72)
	at hudson.model.Queue._withLock(Queue.java:1373)
	at hudson.model.Queue.withLock(Queue.java:1250)
	at hudson.slaves.ComputerRetentionWork.doRun(ComputerRetentionWork.java:63)
	at hudson.triggers.SafeTimerTask.run(SafeTimerTask.java:51)
	at jenkins.security.ImpersonatingScheduledExecutorService$1.run(ImpersonatingScheduledExecutorService.java:58)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.runAndReset(FutureTask.java:308)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$301(ScheduledThreadPoolExecutor.java:180)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:294)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
```
![image](https://user-images.githubusercontent.com/28352910/33731791-59421360-db95-11e7-9962-19caeea31329.png)

**Comment by adveres** [comment]
+1

- CentOS Linux release 7.4.1708 (Core)
- Jenkins ver. 2.93

```
java.lang.ClassCastException: io.jenkins.docker.DockerComputer cannot be cast to hudson.slaves.AbstractCloudComputer
	at hudson.slaves.CloudRetentionStrategy.start(CloudRetentionStrategy.java:43)
	at hudson.model.AbstractCIBase.updateComputer(AbstractCIBase.java:145)
	at hudson.model.AbstractCIBase.access$000(AbstractCIBase.java:46)
	at hudson.model.AbstractCIBase$2.run(AbstractCIBase.java:207)
	at hudson.model.Queue._withLock(Queue.java:1373)
	at hudson.model.Queue.withLock(Queue.java:1250)
	at hudson.model.AbstractCIBase.updateComputerList(AbstractCIBase.java:190)
	at jenkins.model.Jenkins.updateComputerList(Jenkins.java:1548)
	at jenkins.model.Nodes$8.run(Nodes.java:333)
	at hudson.model.Queue._withLock(Queue.java:1373)
	at hudson.model.Queue.withLock(Queue.java:1250)
	at jenkins.model.Nodes.load(Nodes.java:324)
	at jenkins.model.Jenkins$14.run(Jenkins.java:3051)
	at org.jvnet.hudson.reactor.TaskGraphBuilder$TaskImpl.run(TaskGraphBuilder.java:169)
	at org.jvnet.hudson.reactor.Reactor.runTask(Reactor.java:282)
	at jenkins.model.Jenkins$5.runTask(Jenkins.java:1065)
	at org.jvnet.hudson.reactor.Reactor$2.run(Reactor.java:210)
	at org.jvnet.hudson.reactor.Reactor$Node.run(Reactor.java:117)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused: org.jvnet.hudson.reactor.ReactorException
	at org.jvnet.hudson.reactor.Reactor.execute(Reactor.java:269)
	at jenkins.InitReactorRunner.run(InitReactorRunner.java:47)
	at jenkins.model.Jenkins.executeReactor(Jenkins.java:1099)
	at jenkins.model.Jenkins.<init>(Jenkins.java:904)
	at hudson.model.Hudson.<init>(Hudson.java:86)
	at hudson.model.Hudson.<init>(Hudson.java:82)
	at hudson.WebAppMain$3.run(WebAppMain.java:233)
Caused: hudson.util.HudsonFailedToLoad
	at hudson.WebAppMain$3.run(WebAppMain.java:250)
```

**Comment by kaneg** [comment]
me too. Jenkins failed to start after enabling the plugin:

java.lang.ClassCastException: io.jenkins.docker.DockerComputer cannot be cast to hudson.slaves.AbstractCloudComputer
	at hudson.slaves.CloudRetentionStrategy.start(CloudRetentionStrategy.java:43)
	at hudson.model.AbstractCIBase.updateComputer(AbstractCIBase.java:145)
	at hudson.model.AbstractCIBase.access$000(AbstractCIBase.java:46)
	at hudson.model.AbstractCIBase$2.run(AbstractCIBase.java:207)
	at hudson.model.Queue._withLock(Queue.java:1373)
	at hudson.model.Queue.withLock(Queue.java:1250)
	at hudson.model.AbstractCIBase.updateComputerList(AbstractCIBase.java:190)
	at jenkins.model.Jenkins.updateComputerList(Jenkins.java:1548)
	at jenkins.model.Nodes$8.run(Nodes.java:333)
	at hudson.model.Queue._withLock(Queue.java:1373)
	at hudson.model.Queue.withLock(Queue.java:1250)
	at jenkins.model.Nodes.load(Nodes.java:324)
	at jenkins.model.Jenkins$14.run(Jenkins.java:3087)
	at org.jvnet.hudson.reactor.TaskGraphBuilder$TaskImpl.run(TaskGraphBuilder.java:169)
	at org.jvnet.hudson.reactor.Reactor.runTask(Reactor.java:282)
	at jenkins.model.Jenkins$5.runTask(Jenkins.java:1065)
	at org.jvnet.hudson.reactor.Reactor$2.run(Reactor.java:210)
	at org.jvnet.hudson.reactor.Reactor$Node.run(Reactor.java:117)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused: org.jvnet.hudson.reactor.ReactorException
	at org.jvnet.hudson.reactor.Reactor.execute(Reactor.java:269)
	at jenkins.InitReactorRunner.run(InitReactorRunner.java:47)
	at jenkins.model.Jenkins.executeReactor(Jenkins.java:1099)
	at jenkins.model.Jenkins.<init>(Jenkins.java:904)
	at hudson.model.Hudson.<init>(Hudson.java:86)
	at hudson.model.Hudson.<init>(Hudson.java:82)
	at hudson.WebAppMain$3.run(WebAppMain.java:233)
Caused: hudson.util.HudsonFailedToLoad
	at hudson.WebAppMain$3.run(WebAppMain.java:250)

**Comment by niQo** [comment]
same problem on Jenkins 2.89.1 and docker-plugin 1.1, fail at startup
rollback to the previous version of the docker-plugin solved the problem

**Comment by ndeloof** [comment]
short terms workaround : use the "Use Once" strategy for your docker agent template

**Comment by MrZoidberg** [comment]
same problem. @ndeloof will it be fixed in 1.1.1 ?

**Comment by ndeloof** [comment]
@MrZoidberg yes, fixed in 1.1.1 (DockerCloudRetentionStrategy is automatically migrated to DockerOnceRetentionStrategy)

**Comment by adveres** [comment]
Thank you, @ndeloof!
